aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -385,8 +386,14 @@ Tensor & index_copy_(Tensor & self, int64_t dim, const Tensor & index, const Ten
                    source.dim(), "), destination dimensionality (", self.dim(), ")");
   }
 
+  // Should this be TORCH_CHECK?
   TORCH_CHECK_INDEX(index.scalar_type() == ScalarType::Long, "index_copy_(): Expected LongTensor for index");
```

Comment:
Yep, this should be TORCH_CHECK

Also "LongTensor" is antiquated. Expected a "long tensor for index, but got [dtype]" is more modern.